systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/ListenersST.java

```diff
@@ -440,6 +443,210 @@ void testSendMessagesCustomListenerTlsScramSha() {
+            @Step(value = "Create a Kafka cluster with custom listener using TLS authentication " +
+                "and both custom CA certs defined via 'ssl.truststore.location'. " +
+                "Kafka also contains simple authorization config with superuser 'pepa'.", expected = "Kafka cluster with custom listener is ready."),
+            @Step(value = "Create a Kafka topic and Kafka TLS users wit respective ACL configuration.", expected = "Kafka topic and users are created."),
+            @Step(value = "Transmit messages over TLS to custom listener with user-1 certs generated during the firs step.", expected = "Messages are transmitted successfully."),
+            @Step(value = "Transmit messages over TLS to custom listener with Strimzi certs generated for KafkaUser.", expected = "Producer/consumer time-outed due to wrong certificate used."),
+            @Step(value = "Transmit messages over TLS to custom listener with user-2 certs generated during the firs step.", expected = "Messages are transmitted successfully."),
+            @Step(value = "Remove 'ssl.principal.mapping.rules' configuration from Kafka's listener.", expected = "Rolling update of Kafka brokers is performed successfully."),
+            @Step(value = "Transmit messages over TLS to custom listener with user-1 certs generated during the firs step.", expected = "Producer/consumer time-outed due to not-authorized error as KafkaUser CN doesn't match."),
+        },
+        labels = {
+            @Label(value = TestDocsLabels.KAFKA)
+        }
+    )
+    void testSendMessagesCustomListenerTlsCustomization() {
+        final TestStorage testStorage = new TestStorage(ResourceManager.getTestContext());
+
+        String superuserName = "pepa";
+        String customCaCertName = "custom-ca";
+        String customUserCertName1 = "custom-user-1-cert";
+        String customUserCertName2 = "custom-user-2-cert";
+        String mountPath = "/mnt/kafka/custom-authn-secrets/my-listener";
+
+        final CertAndKey rootCa1 = generateRootCaCertAndKey();
+        final CertAndKey rootCa2 = generateRootCaCertAndKey();
+        final CertAndKey user1 = generateEndEntityCertAndKey(rootCa1,
+            SystemTestCertGenerator.retrieveKafkaBrokerSANs(testStorage),
+            "C=CZ, L=Prague, O=Strimzi, CN=" + testStorage.getUsername());
+        final CertAndKey user2 = generateEndEntityCertAndKey(rootCa2,
+            SystemTestCertGenerator.retrieveKafkaBrokerSANs(testStorage),
+            "C=CZ, L=Prague, O=Strimzi, CN=" + superuserName);
+
+        final CertAndKeyFiles rootCertAndKey = exportToPemFiles(rootCa1, rootCa2);
+        final CertAndKeyFiles chainCertAndKey1 = exportToPemFiles(user1);
+        final CertAndKeyFiles chainCertAndKey2 = exportToPemFiles(user2);
+
+        SecretUtils.createCustomCertSecret(testStorage.getNamespaceName(), testStorage.getClusterName(), customUserCertName1, chainCertAndKey1, "user");
```

Comment:
```suggestion
        SecretUtils.createCustomCertSecret(testStorage.getNamespaceName(), testStorage.getClusterName(), customUserCertName1, chainCertAndKey1, testStorage.getUsername());
```

Reply:
`user` is key in secret data, not the name. It has to be like this as clients will look for `user.crt` and `user.key`

Reply:
Okay, so can you make that variable, `secretKey`, at least? I understand that usage, but for others it might be a magic string... where one has to go into method implementation and what that string does.

Reply:
sure